WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -376,8 +376,13 @@ private extension CardReaderConnectionController {
             onError: { [weak self] error in
                 guard let self = self else { return }
 
-                self.analyticsTracker.discoveryFailed(error: error)
-                self.state = .discoveryFailed(error)
+                if case .cancel = self.state {
+                    self.analyticsTracker.connectionFailed(error: error, cardReaderModel: self.candidateReader?.readerType.model)
```

Comment:
I don't understand why we track `connectionFailed` here – it hasn't failed, the user cancelled it.

When connecting in the payments flow, we also track `card_present_collect_payment_canceled`, which is currently being tracked twice for a cancellation (I'm sure that's unrelated to this change. That's not tracked in the `Manage Card Reader` flow though.

Reply:
I added the `connectionFailed` event it seemed the closest fit from existing track events to the manual disconnection:
```
Tracked when connecting to a card reader fails
```
 Up until now was being tracked as `discoveryFailed`, which doesn't seem to fit either. Would you say we should add a new event to track the cancellation tap, or it's not worth it? We do in some other places, like `cardReaderSoftwareUpdateCancelTapped`

Reply:
Reverted with https://github.com/woocommerce/woocommerce-ios/pull/11230/commits/9c5fbcb1135ddf2a50d7006a2e7f828135bb681c, since we're leaving this class as was initially.